src/test/scala/io/iohk/ethereum/consensus/validators/RestrictedEthashBlockHeaderValidatorSpec.scala

```diff
@@ -0,0 +1,149 @@
+    )
+    def createBlockchainConfig(allowedMiners: Set[ByteString]): BlockchainConfig = {
+      BlockchainConfig(
+        frontierBlockNumber = 0,
+        homesteadBlockNumber = 1150000,
+        difficultyBombPauseBlockNumber = 3000000,
+        difficultyBombContinueBlockNumber = 5000000,
+        difficultyBombRemovalBlockNumber = 5900000,
+        byzantiumBlockNumber = 4370000,
+        constantinopleBlockNumber = 7280000,
+        istanbulBlockNumber = 9069000,
+        daoForkConfig = None,
+        // unused
+        maxCodeSize = None,
+        eip155BlockNumber = Long.MaxValue,
+        eip160BlockNumber = Long.MaxValue,
+        eip161BlockNumber = Long.MaxValue,
+        eip150BlockNumber = Long.MaxValue,
+        eip106BlockNumber = 0,
+        chainId = 0x3d.toByte,
+        networkId = 1,
+        monetaryPolicyConfig = null,
+        customGenesisFileOpt = None,
+        accountStartNonce = UInt256.Zero,
+        bootstrapNodes = Set(),
+        gasTieBreaker = false,
+        ethCompatibleStorage = true,
+        atlantisBlockNumber = Long.MaxValue,
+        aghartaBlockNumber = Long.MaxValue,
+        phoenixBlockNumber = Long.MaxValue,
+        petersburgBlockNumber = Long.MaxValue,
+        ecip1098BlockNumber = Long.MaxValue,
+        treasuryAddress = Address(0),
+        ecip1097BlockNumber = Long.MaxValue,
+        checkpointPubKeys = Set.empty,
+        allowedMinersPublicKeys = allowedMiners,
+        ecip1099BlockNumber = Long.MaxValue
+      )
+    }
+    val validParent = BlockHeader(
+      parentHash = ByteStringUtils.string2hash("c12a822d0c9a1a777cd1023172ec304aca76e403355e4eb56592d299e4b86503"),
+      ommersHash = ByteStringUtils.string2hash("1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347"),
+      beneficiary = ByteStringUtils.string2hash("0011223344556677889900112233445566778899"),
+      stateRoot = ByteStringUtils.string2hash("e3a3e62598cdb02a3551f9e932ed248a741ca174c00d977a56d9bb2c6473dd34"),
+      transactionsRoot =
+        ByteStringUtils.string2hash("56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"),
+      receiptsRoot = ByteStringUtils.string2hash("56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"),
+      logsBloom = ByteStringUtils.string2hash("00" * 256),
+      difficulty = BigInt("131520"),
+      number = 10,
+      gasLimit = 5030,
+      gasUsed = 0,
+      unixTimestamp = 1605514463,
+      extraData = ByteStringUtils.string2hash(
+        "6d616e746973808fc245b89183f28ac985019992f202a73c7ab600b0aefa18dcba71a8f3576129280d56f4f499e7a8a53a047e91d73d881745b7a6ac7ca9449fc2b3bb1608921c"
+      ),
+      mixHash = ByteStringUtils.string2hash("2db10efede75cfe87b6f378d9b03e712098e8cd3759784db56d65cc9e9911675"),
+      nonce = ByteStringUtils.string2hash("a57246871d5c8bcc")
+    )
+
+    val validHeader = BlockHeader(
```

Comment:
Another hardcoded header :confused: can we use one of the Fixtures ones and maybe extend it with a signature?

Reply:
nope, if we extend the signature the pow validation will fail. We could mock pow validation, but the whole point of this test is to test the whole validation and pow validation of extended extraData filed. I will add comment there about the fact that those are special headers (with extended extraData field) and they should only be used in those tests.